graylog2-server/src/test/java/org/graylog/plugins/sidecar/collectors/ConfigurationServiceTest.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.plugins.sidecar.collectors;
+
+import com.lordofthejars.nosqlunit.annotation.CustomComparisonStrategy;
+import com.lordofthejars.nosqlunit.annotation.IgnorePropertyValue;
+import com.lordofthejars.nosqlunit.annotation.ShouldMatchDataSet;
+import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
+import com.lordofthejars.nosqlunit.core.LoadStrategyEnum;
+import com.lordofthejars.nosqlunit.mongodb.InMemoryMongoDb;
+import com.lordofthejars.nosqlunit.mongodb.MongoFlexibleComparisonStrategy;
+import org.graylog.plugins.sidecar.database.MongoConnectionRule;
+import org.graylog.plugins.sidecar.rest.models.Configuration;
+import org.graylog.plugins.sidecar.rest.models.NodeDetails;
+import org.graylog.plugins.sidecar.rest.models.Sidecar;
+import org.graylog.plugins.sidecar.services.CollectorService;
+import org.graylog.plugins.sidecar.services.ConfigurationService;
+import org.graylog.plugins.sidecar.services.SidecarService;
+import org.graylog2.bindings.providers.MongoJackObjectMapperProvider;
+import org.graylog2.shared.bindings.ObjectMapperModule;
+import org.graylog2.shared.bindings.ValidatorModule;
+import org.jukito.JukitoRunner;
+import org.jukito.UseModules;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import javax.validation.Validator;
+import java.util.List;
+
+import static com.lordofthejars.nosqlunit.mongodb.InMemoryMongoDb.InMemoryMongoRuleBuilder.newInMemoryMongoDbRule;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(JukitoRunner.class)
```

Comment:
Small nitpicking:

We (almost) only use Jukito in Sidecar tests. I think the only advantage we get is to inject the `ObjectMapperModule` we use in `setUp()` (we also inject a `ValidatorModule` that is never used. I vote for doing like most of our tests and instead of using `JukitoRunner`, create a local instance of `MongoJackObjectMapperProvider` in `setUp()`. Here is an example of that: https://github.com/Graylog2/graylog2-server/blob/269bd12111609d4acad8fcc24c16e7ffb25cf1f5/graylog2-server/src/test/java/org/graylog2/contentpacks/ContentPackInstallationPersistenceServiceTest.java#L54